crates/sui-package-alt/src/sui_flavor.rs

```diff
@@ -150,6 +151,45 @@ impl MoveFlavor for SuiFlavor {
     }
 }
 
+/// We validate that a modern manifest cannot define the "legacy" system names.
+/// This is mainly to protect users
+fn validate_modern_manifest_does_not_use_legacy_system_names(
+    manifest: &ParsedManifest,
+) -> Result<(), String> {
+    // For legacy data, we do not enforce this check.
+    if manifest.legacy_data.is_some() {
+        return Ok(());
+    }
+
+    // collect all manifest deps
+    let mut dep_names = manifest
+        .dependencies
+        .keys()
+        .map(|n| n.get_ref().to_string())
+        .collect::<Vec<_>>();
+
+    // Check "dep replacements" too.
+    dep_names.extend(
+        manifest
+            .dep_replacements
+            .values()
+            .flat_map(|k| k.keys().map(|key| key.to_string()))
+            .collect::<Vec<_>>(),
+    );
+
```

Comment:
You might be able to use `iter.chain` for this, but this way is probably clearer anyhow